release/releaser.go

```diff
@@ -271,23 +296,18 @@ func selectServices(rc *ReleaseContext, spec *flux.ReleaseSpec, results flux.Rel
 		ids := []flux.ServiceID{}
 		for _, s := range spec.ServiceSpecs {
 			if s == flux.ServiceSpecAll {
-				return rc.SelectServices(nil, lockedSet, excludedSet, results, logStatus)
+				return rc.SelectServices(results, logStatus, filtList...)
 			}
 			id, err := flux.ParseServiceID(string(s))
 			if err != nil {
 				return nil, err
 			}
 			ids = append(ids, id)
 		}
-		return rc.SelectServices(ids, lockedSet, excludedSet, results, logStatus)
-	case flux.ServiceSpecAll:
-		return rc.SelectServices(nil, lockedSet, excludedSet, results, logStatus)
+		incFilt := &IncludeFilter{ids}
+		return rc.SelectServices(results, logStatus, append([]ServiceFilter{incFilt}, filtList...)...)
```

Comment:
We need to make sure the include filter runs first. Filters run in order. 

So for example, we need to make sure the include filter is before the lock filter because we want to result in "ignore", even if it is locked.